modules/living_spaces_intranet/living_spaces_intranet.module

```diff
@@ -100,30 +100,43 @@ function living_spaces_intranet_user_access(EntityInterface $entity, $operation,
     $user_spaces = $entity->get('joined_spaces')->getValue();
     $user_spaces = array_column($user_spaces, 'target_id');
 
+    $same = array_intersect($own_spaces, $user_spaces);
     if (
       ($operation === 'delete') &&
       $current->hasPermission('delete shared offices space members')
     ) {
-      $same = array_intersect($own_spaces, $user_spaces);
       foreach ($same as $group_id) {
         /** @var \Drupal\group\Entity\GroupInterface $group */
-        $group = \Drupal::entityTypeManager()->getStorage('group')->load($group_id);
-        if ($group->hasPermission('manage living space member account', $current)) {
+        if (
+          ($group = \Drupal::entityTypeManager()->getStorage('group')->load($group_id))
```

Comment:
no need to hardcode group types. you can just add `$group->hasField()` because other types don't have the field.